docs/BUILD.md

```diff
@@ -0,0 +1,449 @@
+
+* `Gopkg.toml` - with all of the dependencies including the specific version of the package itself as a `constraint`
+* `Gopkg.lock` - generated from a run
+* `dummy.go` - an empty file to give go something to build
+
+To build a specific tool, you can execute `make bin/<tool-name>`, e.g. `make bin/manifest-tool`, in `build-tools`, e.g.:
+
+```
+make -C build-tools bin/manifest-tool
+```
+
+This target does the following:
+
+1. Set `GOPATH=$PWD`, i.e. the `build-tools/` directory
+2. If `dep` isn't available, `go get` it
+3. `cd src/<tool> && dep ensure -v`. This step installs the tool's go source in `vendor/`, as it is in `Gopkg.toml`
+4. `cd src/<tool>/vendor/<path>` ; this path will change based on the tool.
+5. `go build <options> -o $GOPATH/bin/<tool>`
+
+To build all of the tools, run `make -C build-tools all`, or in the project root directory, just `make build-tools`
+
+#### Reasoning
+
+The normal process for installing go-based binaries is to execute `go get` with options, e.g.
+
+```
+go get -u github.com/linuxkit/linuxkit/src/cmd/linuxkit
+go get -u github.com/estesp/manifest-tool
+```
+
+EVE uses a somewhat non-standard build process for these tools to ensure specific versions without polluting the user's normal workspace. 
+
+## Output Components
+
+The following are the output components from the build process and their purpose. There are two kinds of components: final, intended for actual direct usage, and interim, used to build the final components. Some interim components may be removed as part of the build finalization process.
+
+#### Final
+
+* `fallback.img` - a symlink to `fallback.qcow2`
+* `fallback.qcow2` - the final bootable live disk image in [qcow2](https://en.wikipedia.org/wiki/Qcow) format
+* `installer.raw` - a bootable image that can install EVE on a local device. The installer is intended to be flashed to a USB or SD device, or booted via PXE, and then run to install on a local drive.
+* `installer.iso` - a bootable ISO image with a hidden EFI boot partition and an installer partition, with the contents of `installer.raw`. The installer is intended to be booted in a manner typical of iso files, and then run to install on a local drive.
+
+#### Interim
+
+* `rootfs.img` - a live bootable rootfs filesystem image. This can be either [squashfs](https://en.wikipedia.org/wiki/SquashFS) (default) or [ext4](https://en.wikipedia.org/wiki/Ext4). 
+* `rootfs_installer.img` - a bootable rootfs filesystem image to run as an installer.
+* `fallback.raw` - a live bootable disk image, will be converted to [qcow2](https://en.wikipedia.org/wiki/Qcow). Has 2 gpt partitions:
+    1. UEFI partition with grub
+    2. root partition from the above `rootfs.img`
+* `config.img` - 1MB FAT32 image file containing basic configuration information, including wpa supplicant, name of the controller, onboarding key/cert pair, and other configuration information. 
```

Comment:
I'd say it contains identity and trust information, plus some config information needed to deal with network conditions like using static IP addresses and enterprise proxies; things that are needed in order to the controller.

Reply:
Wouldn't those change from install to install? Right now it appears to be a static set of things (although I notice it is called `maketestconfig.sh`, emphasis on "test"). Does that mean that we have a more dynamic mechanism for real production deployment? Or we don't have but we plan on one?

Reply:
We have a couple of different approaches for onboarding, and for some the image changes from install to install, and in other cases there is an image per vendor+model plus additional information to tie the shipped box (serial number, qr code with a cert, etc)

For example, the approach we initially called the DUI approach is to create a device in zedcloud, ask zedcloud to generate a single use onboarding token for that device, put the onboarding token in conf/ and build a unique image for that device.

We are likely to come up with more variants at Zededa since it depends on what the different factories can do when it comes to pre-install software and get or set identities (for instance, we don't yet know whether any factory can extract the TPM identity).

Also, for EVE this depends on what Adam would support; for testing one can do a FCFS approach where anybody (or the first one) connecting with a unique device certificate will get onboarded in Adam.